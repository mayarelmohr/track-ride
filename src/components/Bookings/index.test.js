import React from "react";
import User from "./user";
import Booking from "./index";
import { render } from "@testing-library/react";

const mockBooking = [
  {
    id: 123,
    name: "John Doe",
    picture: "https://robohash.org/1234?size=50x50",
    station: "213",
    payment: "credit",
    status: "in",
  },
];

test("Shows user data correctly", () => {
  const { getByTestId, getByAltText } = render(
    <User booking={mockBooking[0]} />
  );
  const user = getByTestId("user");
  expect(user).toHaveTextContent("checked in");
  expect(user).toHaveTextContent(/213/);
  const image = getByAltText("John Doe");
  expect(image);
});

test("Shows Booking data", () => {
  const { getAllByTestId } = render(<Booking bookingsList={mockBooking} />);
  const bookingItem = getAllByTestId("user");
  expect(bookingItem).toHaveLength(1);
});

test("add snapshot for user", () => {
  const { container, asFragment } = render(<User booking={mockBooking[0]} />);
  expect(asFragment(container)).toMatchSnapshot();
});

test("add snapshot for booking", () => {
  const { container, asFragment } = render(
    <Booking bookingsList={mockBooking} />
  );
  expect(asFragment(container)).toMatchSnapshot();
});
