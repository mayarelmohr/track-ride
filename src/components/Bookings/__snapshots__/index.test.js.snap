// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add snapshot for booking 1`] = `
<DocumentFragment>
  <section>
    <h2>
      Bookings
    </h2>
    <ul
      class="list"
    >
      <li>
        <div
          class="user"
          data-testid="user"
        >
          <div
            class="image"
          >
            <img
              alt="John Doe"
              loading="lazy"
              src="https://robohash.org/1234?size=50x50"
            />
          </div>
          <div>
            <p
              class="name"
            >
              John Doe
            </p>
            <p
              class="info"
            >
              213 checked in
            </p>
            <p
              class="info"
            >
              <span
                aria-label="payment"
                role="img"
              >
                ðŸ’µ credit
              </span>
            </p>
          </div>
        </div>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;

exports[`add snapshot for user 1`] = `
<DocumentFragment>
  <div
    class="user"
    data-testid="user"
  >
    <div
      class="image"
    >
      <img
        alt="John Doe"
        loading="lazy"
        src="https://robohash.org/1234?size=50x50"
      />
    </div>
    <div>
      <p
        class="name"
      >
        John Doe
      </p>
      <p
        class="info"
      >
        213 checked in
      </p>
      <p
        class="info"
      >
        <span
          aria-label="payment"
          role="img"
        >
          ðŸ’µ credit
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
